src/IterableCodeExtractor.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+namespace WP_CLI\I18n;
+
+use Gettext\Translation;
+use Gettext\Translations;
+use RecursiveCallbackFilterIterator;
+use RecursiveDirectoryIterator;
+use RecursiveIteratorIterator;
+use DirectoryIterator;
+use WP_CLI;
+
+trait IterableCodeExtractor {
+
+	private static $dir = '';
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public static function fromFile( $file, Translations $translations, array $options = [] ) {
+		foreach ( static::getFiles( $file ) as $f ) {
+			// Make sure a relative file path is added as a comment.
+			$options['file'] = ltrim( str_replace( static::$dir, '', $f ), '/' );
+
+			$string = file_get_contents( $f );
+
+			if ( ! $string ) {
+				WP_CLI::debug(
+					sprintf(
+						'Could not load file %1s',
+						$f
+					)
+				);
+
+				continue;
+			}
+
+			if ( ! empty ( $options['wpExtractTemplates'] ) ) {
+				$headers = MakePotCommand::get_file_data_from_string( $string, [ 'Template Name' => 'Template Name' ] );
+
+				if ( ! empty( $headers[ 'Template Name'])) {
+					$translation = new Translation( '', $headers[ 'Template Name'] );
+					$translation->addExtractedComment('Template Name of the theme' );
+
+					$translations[] = $translation;
+				}
+			}
+
+			static::fromString( $string, $translations, $options );
+		}
+	}
+
+	/**
+	 * Recursively extracts the translations from a directory.
+	 *
+	 * @param string $dir Root path to start the recursive traversal in.
+	 * @param Translations $translations The translations instance to append the new translations.
+	 * @param array $options
```

Comment:
As noted above, we could describe the possible options a bit better. Perhaps the way WordPress core does it: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/php/#1-1-parameters-that-are-arrays